### PR TITLE
[no-issue]: Split pull API for data-broker Services

### PR DESCRIPTION
**Pull request by kael89**
This PR continues the work that started in https://github.com/beyondessential/tupaia/pull/4704 by splitting the pull API for `data-broker` services into `pullAnalytics()`, `pullEvents()`, `pullSyncGroupsResults()`

### CODE REVIEW

File: packages/data-broker/src/__tests__/DataBroker/DataBroker.test.ts
Comment (kael89) [inline packages/data-broker/src/__tests__/DataBroker/DataBroker.test.ts:314]: Just moving these test cases one level up, under a single `pull permissions` describe block

File: packages/data-broker/src/__tests__/services/superset/SupersetService.test.ts
Comment (kael89) [inline packages/data-broker/src/__tests__/services/superset/SupersetService.test.ts:44]: Lifiting test cases one level up & using `.pullAnalytics()` instead of `.pull()`

File: packages/data-broker/src/__tests__/services/tupaia/TupaiaService.test.ts
Comment (kael89) [inline packages/data-broker/src/__tests__/services/tupaia/TupaiaService.test.ts:120]: I duplicated some test cases for `pullAnalytics` and `pullEvents`, since they are separate APIs now

File: packages/data-broker/src/services/Service.ts
Comment (kael89) [inline packages/data-broker/src/services/Service.ts:65]: I simplified the options type to `Record<string, unknown>`. This is because we shouldn't leak any abstraction details unless we have to - no need to make any assumptions about the actual options used by each service.

File: packages/data-broker/src/services/weather/WeatherService.ts
Comment (kael89) [inline packages/data-broker/src/services/weather/WeatherService.ts:52]: This is the only case where I left the original, data-source-level `pull()` methods in place. Specific logic is repeated across data sources so it kind of makes sense that way.

File: packages/data-broker/src/__tests__/services/dhis/DhisService.test.ts
Comment (kael89) [inline packages/data-broker/src/__tests__/services/dhis/DhisService.test.ts:251]: Duplicating shared test cases between `pullAnalytics` and `pullEvents` and restructuring test cases a bit

**Review by kael89 — COMMENTED** [review]
Most of the diff is duplicating tests cases and moving them one level up, to match the new API structure

File: packages/data-broker/src/__tests__/services/dhis/DhisService.test.ts
Comment (edmofro) [inline packages/data-broker/src/__tests__/services/dhis/DhisService.test.ts:364]:
```suggestion
    it('looks up the api from the given data group', async () => {
```

**Review by edmofro — CHANGES_REQUESTED** [review]
More great stuff from the open source community! 

I've pointed out a couple of spots where we could remove the "data source" term altogether, there are probably a bunch more too?

**Comment by kael89** [comment]
> More great stuff from the open source community!
> 
> I've pointed out a couple of spots where we could remove the "data source" term altogether, there are probably a bunch more too?

Ah great call, I missed a lot of those 👀 . I did a quick scan and renamed a bunch

**Review by edmofro — APPROVED** [review]
Perfect! Again, tysm for this @kael89

**Comment by kael89** [comment]
@edmofro I've done a quick smoke test locally and this works with Tupaia service vizes... Merge?

**Comment by edmofro** [comment]
@kael89 our dev build is broken at the moment, so although it's perfect timing straight after Tupaia release to merge, we should hold off to avoid adding noise to dev while that broken build is being worked on 🙏

**Comment by edmofro** [comment]
Fixed! Merge when ready

**Pull request merged**
